packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -781,21 +770,61 @@ export class ContainerRuntime
 
 		const mc = loggerToMonitoringContext(logger);
 
+		// Some options require a minimum version of the FF runtime to operate, so the default configs will be generated
+		// based on the compatibility mode.
+		// For example, if compatibility mode is set to "1.0.0", the default configs will ensure compatibility with FF runtime
+		// 1.0.0 or later. If the compatibility mode is set to "2.10.0", the default values will be generated to ensure compatibility
+		// with FF runtime 2.10.0 or later.
+		// TODO: We will add in a way for users to pass in compatibilityMode in a follow up PR.
+		const compatibilityMode = defaultCompatibilityMode;
+		if (!isValidCompatMode(compatibilityMode)) {
+			throw new UsageError(
+				`Invalid compatibility mode: ${compatibilityMode}. It must be an existing FF version (i.e. 2.22.1).`,
+			);
+		}
+		const defaultVersionDependentConfigs = getConfigsForCompatMode(compatibilityMode);
+
+		// The following are the default values for the options that do not affect the DocumentSchema.
+		const defaultConfigsNonVersionDependent: Required<
+			Omit<IContainerRuntimeOptionsInternal, keyof DocSchemaAffectingRuntimeOptions>
+		> = {
+			summaryOptions: {},
+			loadSequenceNumberVerification: "close",
+			maxBatchSizeInBytes: defaultMaxBatchSizeInBytes,
+			chunkSizeInBytes: defaultChunkSizeInBytes,
+		};
+
+		const defaultConfigs = {
+			...defaultVersionDependentConfigs,
+			...defaultConfigsNonVersionDependent,
+		};
+
+		// Here we set each option to its corresponding default config value if it's not provided in runtimeOptions.
```

Comment:
And if you miss a default line then the compiler will complain right?  Since the provided one is Partial or something?

Reply:
Could you clarify what you mean by miss a default line?